src/main/java/htsjdk/samtools/util/AbstractRecordAndOffset.java

```diff
@@ -124,41 +113,20 @@ public String getReadName() {
     public byte[] getBaseQualities() {
         return record.getBaseQualities();
     }
-
+    
+    /**
+     * @return the base quality according to <code>offset</code>.
+     */
+    public byte getBaseQuality() {
+        return record.getBaseQualities()[offset];
+    }
 
     /**
      * @param position in the reference
      * @return base quality of a read base, corresponding to a given reference position
      */
     public byte getBaseQuality(int position) {
-        int index = getRelativeOffset(position);
-        byte[] baseQualities = record.getBaseQualities();
-        validateIndex(index, baseQualities);
-        return baseQualities[index];
-    }
-
-    private void validateIndex(int index, byte[] array) {
-        if (index < 0 || index >= array.length) {
-            throw new IllegalArgumentException("The requested position is not covered by this AbstractRecordAndOffset" +
-                    " object.");
-        }
-    }
-
-    private int getRelativeOffset(int position) {
-        if (refPos == -1) {
-            return position - offset;
-        }
-        return position - refPos + offset;
-    }
-
-    @Override
-    public int hashCode() {
-        if (hash != 0) return hash;
-        hash = record.hashCode();
-        hash = 31 * hash + length;
-        hash = 31 * hash + offset;
-        hash = 31 * hash + refPos;
-        return hash;
+        return record.getBaseQualities()[position - offset];
```

Comment:
why is position-offset a reasonable number to look into the BaseQ array? could it not be negative? or too large?